src/vehicle_base.h

```diff
@@ -749,6 +749,8 @@ struct Vehicle : VehiclePool::PoolItem<&_vehicle_pool>, BaseVehicle, BaseConsist
 
 		this->profit_this_year = src->profit_this_year;
 		this->profit_last_year = src->profit_last_year;
+
+		GroupStatistics::AddProfitLastYear(this);
```

Comment:
I don't see anything removing profit, so why would autoreplace/autorenew add profit to the statistics? It's not removed by deleting the old vehicle as far as I can see. So this looks like you're going to double count the profits of last year.
At best you will create a difference in the statistics between a game that is loaded from savegame after the autrenew and a game that has been running during the autorenew. At worst this will cause a desync if something uses these `GroupStatistics` in the main game logic.

Reply:
I created a debug branch to test the autoreplace issue: https://github.com/SamuXarick/OpenTTD/tree/group-get-profit-last-year-debug

Savegame: [autoreplace test.zip](https://github.com/OpenTTD/OpenTTD/files/5821631/autoreplace.test.zip)
AI: [autoreplace test.zip](https://github.com/OpenTTD/OpenTTD/files/5821636/autoreplace.test.zip)

How to debug test:
1 - Savegame contains the test AI that is constantly calling AIGroup.GetProfitLastYear.
2 - AIGroup.GetProfitLastYear has two methods for retrieving the last year profit. One computes manually by iterating all vehicles in the group, the other retrieves directly from the group statistics last year profit. It compares the result of both if they match with an assert.
3 - Load the savegame and fastforward, wait for the vehicle to become old and trigger the autorenew.

Reply:
https://github.com/OpenTTD/OpenTTD/blob/921c6591f95d3c46432e7c3c7cc48ac8a629d9c3/src/train_cmd.cpp#L1419-L1427

Okay, I see what you mean. I'm investigating if there's another way to update the profit_last_year.

Reply:
And it is now fixed again. I moved it away from `CopyVehicleConfigAndStatistics` in `vehicle_base.h` to 
`CopyHeadSpecificThings` in `autoreplace_cmd.cpp`

https://github.com/OpenTTD/OpenTTD/blob/dd6bf061cdb0305a993bcdfe764a966ffb8e1029/src/autoreplace_cmd.cpp#L425